integrations/optimum/pyproject.toml

```diff
@@ -0,0 +1,194 @@
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+name = "optimum-haystack"
+dynamic = ["version"]
+description = "Component to embed strings and Documents using models loaded with the HuggingFace Optimum library. This component is designed to seamlessly inference models using the high speed ONNX runtime."
+readme = "README.md"
+requires-python = ">=3.8"
+license = "Apache-2.0"
+keywords = []
+authors = [
+  { name = "deepset GmbH", email = "info@deepset.ai" },
+  { name = "Ashwin Mathur", email = "" },
+]
+classifiers = [
+  "Development Status :: 4 - Beta",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+dependencies = [
+  "haystack-ai",
+  "transformers[sentencepiece]==4.36.2",
+  "sentence-transformers>=2.2.0",
```

Comment:
I think we can remove the `sentence-transformers` dependency. I ran the tests locally with this removed and everything still passed.

Reply:
Or is there a scenario in which this dependency is needed?

Reply:
I took a deeper dive into this, since I found I was getting errors like this 
```python
../../../../../../Library/Application Support/hatch/env/virtual/optimum-haystack/oY5Xs40Q/optimum-haystack/lib/python3.10/site-packages/optimum/exporters/tasks.py:1240: in get_model_class_for_task
    loaded_library = importlib.import_module(library)
/opt/homebrew/Caskroom/miniforge/base/envs/haystack-core-integrations/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'sentence_transformers', import_ = <function _gcd_import at 0x10115f490>

>   ???
E   ModuleNotFoundError: No module named 'sentence_transformers'
```
when running the tests with versions `transformers[sentencepiece]>=4.37.2` and `optimum[onnxruntime]>=1.16.2`. 

The culprit it appears is that the [main export function](https://github.com/huggingface/optimum/blob/8651c0ca1cccf095458bc80329dec9df4601edb4/optimum/exporters/onnx/__main__.py#L164) of Optimum into ONNX has a option called `library_name`
```python
        library_name (`Optional[str]`, defaults to `None`):
            The library of the model (`"transformers"` or `"diffusers"` or `"timm"` or `"sentence_transformers"`). If not provided, will attempt to automatically detect the library name for the checkpoint.
``` 
which auto detects that some of the models in the test should be loaded with the `sentence_transformers` library. I'm not entirely sure why this error doesn't occur with the current pinned dependencies (I was able to get tests to pass without sentence transformers installed), but it looks like Optimum basically has some hidden dependencies (namely these three additional libraries `diffusers`, `timm` and `sentence_transformers`) depending on which model you load from the HF Hub. 

Not entirely sure how to proceed here, but it seems like keeping `sentence_transformers` as a dependency is probably a good idea due to the popularity of the format.

Reply:
You're right, this is the reason I added `sentence_transformers` as a dependency.

Reply:
@shadeMe should we add this as a developer comment somewhere? I'm thinking that future users or devs might run into this issue for `timm` or `diffusers` (or any other libraries they might add) and be confused.

Reply:
Definitely! We can directly add it to `pyproject.toml`, I think?

Reply:
Sounds good! I will add it to these comments to the `pyproject.toml`.